src/nomad_measurements/xrd/schema.py

```diff
@@ -889,16 +792,21 @@ def write_nx_xrd(
         )
         merge_sections(self, xrd, logger)
 
-        nexus_output = None
-        if self.generate_nexus_file:
-            archive_name = archive.metadata.mainfile.split('.')[0]
-            nexus_output = f'{archive_name}_output.nxs'
-        handle_nexus_subsection(
-            xrd_dict,
-            nexus_output,
-            archive,
-            logger,
-        )
+        
+        archive_name = archive.metadata.mainfile.split('.')[0]
+        nexus_output = f'{archive_name}.nxs'
+        dataconverter.logger = logger
+        with archive.m_context.raw_file(self.data_file) as file:
+            dataconverter.convert(input_file=[file.name], output=nexus_output, reader="xrd", nxdl="NXxrd_pan", skip_verify=True)
+        # TODO: Need to check what NeXus definitions version can parse this NXxrd_pan
+        # from nomad.parsing.nexus.nexus import NexusParser
+        # nexus_parser = NexusParser()
+        # nexus_parser.parse(mainfile=nexus_output, archive=archive, logger=logger)
```

Comment:
I am confused there, how did you get the you got the template from `pynxtools_xrd.XRDReader`, again processing the raw file by nexus workflow from nomad, which generating the `Template` as a side product of the workflow. Therefor taking time twice for the same task.

Reply:
Indeed. It does it twice. This is an optional flag so I didn't really care. This works with all correct logging etc. We can also just use the writer if we wanna optimize. But we're duplicating data all anyways.